app/livechat/server/hooks/beforeCloseRoom.js

```diff
@@ -0,0 +1,22 @@
+import { Meteor } from 'meteor/meteor';
+
+import { callbacks } from '../../../callbacks';
+import { LivechatDepartment } from '../../../models';
+
+callbacks.add('livechat.beforeCloseRoom', (room) => {
+	const { departmentId } = room;
+	if (!departmentId) {
+		return room;
+	}
+
+	const department = LivechatDepartment.findOneById(departmentId);
+	if (!department || !department.tagRequiredWhenClosingChat) {
```

Comment:
the field you're saving is called `requestTagBeforeClosingChat` but here you're testing the field `tagRequiredWhenClosingChat` .. so it is not working.